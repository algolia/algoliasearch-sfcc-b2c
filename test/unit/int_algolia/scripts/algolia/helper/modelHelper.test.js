const MasterProductMock = require('../../../../../mocks/dw/catalog/MasterProduct');

const modelHelper = require('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/helper/modelHelper');
const VariantMock = require("../../../../../mocks/dw/catalog/Variant");
const collectionHelper = require("../../../../../mocks/helpers/collectionHelper");

describe('getColorVariations', () => {
    test('en locale', () => {
        const masterProduct = new MasterProductMock();
        const colorVariations = modelHelper.getColorVariations(masterProduct, 'en');
        expect(colorVariations).toMatchSnapshot();
    });
    test('fr locale', () => {
        const masterProduct = new MasterProductMock();
        const colorVariations = modelHelper.getColorVariations(masterProduct, 'fr');
        expect(colorVariations).toMatchSnapshot();
    });
});

test('generateVariantRecordsWithColorVariations', () => {
    const masterProduct = new MasterProductMock();
    const variantPinkSize4 = new VariantMock({
        ID: '701644031206M',
        variationAttributes: { color: 'JJB52A0', size: '004' },
        masterProduct,
    });
    const variantPinkSize6 = new VariantMock({
        ID: '701644031213M',
        variationAttributes: { color: 'JJB52A0', size: '006' },
        masterProduct,
    });
    masterProduct.variants = collectionHelper.createCollection([
        variantPinkSize4,
        variantPinkSize6,
    ]);

    const variantRecords = modelHelper.generateVariantRecordsWithColorVariations({
        masterProduct,
        locales: collectionHelper.createCollection(['fr']),
        attributeList: ['name', 'primary_category_id', 'categories', 'in_stock', 'price', 'url'],
        nonLocalizedAttributeList: [],
    });
    expect(variantRecords).toMatchSnapshot();
});
