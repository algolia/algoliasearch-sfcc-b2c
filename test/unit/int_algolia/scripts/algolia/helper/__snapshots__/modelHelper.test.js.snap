// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getColorVariations en locale 1`] = `
[
  {
    "color": "Hot Pink Combo",
    "image_groups": [
      {
        "_type": "image_group",
        "images": [
          {
            "_type": "image",
            "alt": "Floral Dress, Hot Pink Combo, large",
            "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dwcc434d54/images/large/PG.10237222.JJB52A0.PZ.jpg",
            "title": "Floral Dress, Hot Pink Combo",
          },
          {
            "_type": "image",
            "alt": "Floral Dress, Hot Pink Combo, large",
            "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw58a034a4/images/large/PG.10237222.JJB52A0.BZ.jpg",
            "title": "Floral Dress, Hot Pink Combo",
          },
        ],
        "view_type": "large",
      },
      {
        "_type": "image_group",
        "images": [
          {
            "_type": "image",
            "alt": "Floral Dress, Hot Pink Combo, small",
            "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw4e4ce4f6/images/small/PG.10237222.JJB52A0.PZ.jpg",
            "title": "Floral Dress, Hot Pink Combo",
          },
          {
            "_type": "image",
            "alt": "Floral Dress, Hot Pink Combo, small",
            "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw2612fb5e/images/small/PG.10237222.JJB52A0.BZ.jpg",
            "title": "Floral Dress, Hot Pink Combo",
          },
        ],
        "view_type": "small",
      },
      {
        "_type": "image_group",
        "images": [
          {
            "_type": "image",
            "alt": "Floral Dress, Hot Pink Combo, swatch",
            "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw506aa734/images/swatch/PG.10237222.JJB52A0.CP.jpg",
            "title": "Floral Dress, Hot Pink Combo",
          },
        ],
        "view_type": "swatch",
      },
    ],
    "variationURL": "/on/demandware.store/Sites-Algolia_SFRA-Site/en/Product-Show?pid=25592581M&dwvar_25592581M_color=JJB52A0",
  },
]
`;

exports[`getColorVariations fr locale 1`] = `
[
  {
    "color": "Combo rose vif",
    "image_groups": [
      {
        "_type": "image_group",
        "images": [
          {
            "_type": "image",
            "alt": "Robe florale, Combo rose vif, large",
            "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dwcc434d54/images/large/PG.10237222.JJB52A0.PZ.jpg",
            "title": "Robe florale, Combo rose vif",
          },
          {
            "_type": "image",
            "alt": "Robe florale, Combo rose vif, large",
            "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw58a034a4/images/large/PG.10237222.JJB52A0.BZ.jpg",
            "title": "Robe florale, Combo rose vif",
          },
        ],
        "view_type": "large",
      },
      {
        "_type": "image_group",
        "images": [
          {
            "_type": "image",
            "alt": "Robe florale, Combo rose vif, small",
            "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw4e4ce4f6/images/small/PG.10237222.JJB52A0.PZ.jpg",
            "title": "Robe florale, Combo rose vif",
          },
          {
            "_type": "image",
            "alt": "Robe florale, Combo rose vif, small",
            "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw2612fb5e/images/small/PG.10237222.JJB52A0.BZ.jpg",
            "title": "Robe florale, Combo rose vif",
          },
        ],
        "view_type": "small",
      },
      {
        "_type": "image_group",
        "images": [
          {
            "_type": "image",
            "alt": "Robe fleurie, Mix rose vif, swatch",
            "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw506aa734/images/swatch/PG.10237222.JJB52A0.CP.jpg",
            "title": "Robe fleurie, Mix rose vif",
          },
        ],
        "view_type": "swatch",
      },
    ],
    "variationURL": "/on/demandware.store/Sites-Algolia_SFRA-Site/fr/Product-Show?pid=25592581M&dwvar_25592581M_color=JJB52A0",
  },
]
`;
